server/src/main/java/org/elasticsearch/common/logging/JULBridge.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.logging;
+
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.logging.Handler;
+import java.util.logging.LogRecord;
+import java.util.stream.Collectors;
+
+/**
+ * A Java Util Logging handler that writes log messages to log4j.
+ */
+class JULBridge extends Handler {
+
+    private static final Map<java.util.logging.Level, Level> levelMap = Map.of(
+        java.util.logging.Level.OFF,
+        Level.OFF,
+        java.util.logging.Level.SEVERE,
+        Level.ERROR,
+        java.util.logging.Level.WARNING,
+        Level.WARN,
+        java.util.logging.Level.INFO,
+        Level.INFO,
+        java.util.logging.Level.FINE,
+        Level.DEBUG,
+        java.util.logging.Level.FINEST,
+        Level.TRACE,
+        java.util.logging.Level.ALL,
+        Level.ALL
+    );
+
+    private static final TreeMap<Integer, Level> sortedLevelMap = new TreeMap<>(
+        levelMap.entrySet().stream().collect(Collectors.toMap(e -> e.getKey().intValue(), Map.Entry::getValue))
+    );
+
+    public static void install() {
+        var rootJulLogger = java.util.logging.LogManager.getLogManager().getLogger("");
+        // clear out any other handlers, so eg we don't also print to stdout
+        for (var existingHandler : rootJulLogger.getHandlers()) {
```

Comment:
is it a good idea to remove elements while executing the loop?

Reply:
It's an array, so it doesn't actually change as remove is called.

Reply:
should be safe. the getHandlers return a copied array of handlers